### PR TITLE
 feat #192: V2 Add an interface to use extra-model inside of Python

### PR DESCRIPTION
**Pull request by rushic24**
Follow up from my old PR here https://github.com/wayfair-incubator/extra-model/pull/198

Let me know if you mean this structure, I can just add the tests and other docstrings later.

just adding the checklist from previous PR

- [x] please, add docstring to the function (inputs/outputs are documented here - https://wayfair-incubator.github.io/extra-model/site/#extra-model-input and https://wayfair-incubator.github.io/extra-model/site/#extra-model-output)
- [x] add tests for the new function (1-2 is enough, there is an example in there already that you can reference)
- [x] add info about the change to CHANGELOG
- [x] you've correctly identified the function that needed to be refactored out of run, but now I need you to change run so that we don't duplicate code for no reason.
- [x] make sure that linters pass
- [x] you need to add this option to this file - https://wayfair-incubator.github.io/extra-model/site/running_extra/ (with examples).
- [x] also feel free to upload your jupyter notebook here - https://github.com/wayfair-incubator/extra-model/tree/main/docs/examples and then you can reference it above

### CODE REVIEW

**Comment by romatik** [comment]
just adding the checklist from previous PR 

- [ ] please, add docstring to the function (inputs/outputs are documented here - https://wayfair-incubator.github.io/extra-model/site/#extra-model-input and https://wayfair-incubator.github.io/extra-model/site/#extra-model-output)
- [ ] add tests for the new function (1-2 is enough, there is an example in there already that you can reference)
- [ ] add info about the change to CHANGELOG
- [ ] you've correctly identified the function that needed to be refactored out of `run`, but now I need you to change `run` so that we don't duplicate code for no reason.
- [ ] make sure that linters pass
- [ ] you need to add this option to this file - https://wayfair-incubator.github.io/extra-model/site/running_extra/ (with examples). 
- [ ] also feel free to upload your jupyter notebook here - https://github.com/wayfair-incubator/extra-model/tree/main/docs/examples and then you can reference it above

File: extra_model/_run.py [hunk 1]
```diff
@@ -21,24 +41,32 @@ def run(
...
-        raise ExtraModelError(
-            f"Input columns must include `CommentId` and `Comments`, \
-        but got {input_data.columns.to_list()} instead"
-        )
 
-    logger.info("Running `extra-model`")
-    results_raw = extra_model.predict(comments=input_data.to_dict("records"))
-    results = pd.DataFrame(results_raw)
+    check_input_structure(input_data)
+    results = build_model_and_predict(embeddings_path, input_data)
```
Comment (romatik) [inline extra_model/_run.py:48]: then this will be only `results = run_from_dataframe(input_data, embeddings_path)`

**Review by romatik — CHANGES_REQUESTED** [review]
This is almost 100% what I had in mind, thanks for working on this

**Comment by rushic24** [comment]
would ` you need to add this option to this file - https://wayfair-incubator.github.io/extra-model/site/running_extra/ (with examples).` this be needed? This function will only be useful inside python files.

**Comment by romatik** [comment]
> would ` you need to add this option to this file - https://wayfair-incubator.github.io/extra-model/site/running_extra/ (with examples).` this be needed? This function will only be useful inside python files.

Yes, this is needed as well. It's another interface that people will be able to use, so we need to document it.

File: tests/test_run.py [hunk 2]
```diff
@@ -71,3 +71,13 @@ def test_run__wrong_input_raises_error(mocker, extra_model_mock, pandas_mock):
 
     with pytest.raises(ExtraModelError, match="wrong_column_1"):
         run(mocker.MagicMock(), mocker.MagicMock())
+
+
+def test_run__run_from_dataframe(mocker, extra_model_mock, pandas_mock):
```
Comment (romatik) [inline tests/test_run.py:76]: can you also add one more test that passes wrong `df` to make sure we trigger the error? There is an example of it already above this test.

Comment (rushic24) [reply tests/test_run.py:76]: Added, run was using run_from_dataframe so that is why it showed 100% coverage even before

**Comment by rushic24** [comment]
Can we have auto run workflows after a git push?

**Comment by romatik** [comment]
you can run it locally with `docker-compose run test`, but yeah, the settings for first-time contributors is that by default your commits don't trigger actions.

**Review by romatik — APPROVED** [review]

**Comment by romatik** [comment]
I'll go over the PR tomorrow and merge it to `main`

**Comment by romatik** [comment]
thanks a lot for working on this!

**Comment by rushic24** [comment]
NP, I just came across this repo and looked like completely new so thought of going deeper. I didn't know much about PyTest, got a chance get my hands on it :)

**Pull request merged**